WorkingInR/3_Importing_Data_in_R.Rmd

```diff
@@ -2,61 +2,252 @@
+Accordingly, the object `weather2`, now visible in the tab `Environment` (see figure \@ref(fig:csv-env2)), is described as a tibble with 12 observations and only one variable.
+
+``` {r csv-env2, echo=FALSE, fig.cap="_The tibble_ weather2 _is now stored, but is wrongly formatted._", out.width="100%"}
+knitr::include_graphics("Pics/csv-env2.png")
+```
+
+The content of the resulting tibble is as follows:
+```{r weather2, echo=TRUE} 
+weather2
+```
 
-## read_csv()
 
-In a _*C*omma *S*eparated *V*alues_ file, the data is delimited by a comma `,` (hence the name). It is possibly the most common type 
+This shows how important it is to review the data before importing it. 
+A quick check would have revealed the use of `;` as delimiter in `weather2.csv`, and you would have certainly used the follow code to import it. 
 
-## read_csv2()
-In countries (such as most of the European countries) where a comma is used as a decimal separator, the data is delimited by a semi-colon `;`. 
+```{r delim-csv2bis, echo=TRUE} 
+weather2 <- read_delim(file = "Files/weather2.csv", delim = ";")
+```
+Here, the message `Column specification` tells you that the content of the columns `month` and `Wind (m/s)` has been recognized as of data type `character`, and that the two other columns have been recognized as `double`. While `read_delim()` got things right about the number of variables, something went wrong with the variables as we could expect `Wind (m/s)` to be recognized as `double`.
+To find out about this issue we need to review the imported data in the object `weather2` and compare it to the original file `weather2.csv`.
+
+### Review the imported data
+
+The example above shows the importance of carefully verifying that the imported data in R matches the original data set.
+The following figure compares the content of the file `weather2.csv` (Fig. \@ref(fig:comp) left) to the content of the object `weather2` (Fig. \@ref(fig:comp) right):
+
+``` {r comp, echo=FALSE, fig.cap="_The data in the object does not match the data in the file._", out.width="100%"}
+knitr::include_graphics("Pics/comp.png")
+```
 
-## read_excel()
+Looking at the rows in both pictures, one can understand that all commas have simply been ignored, excepted those in the last column.
+This is a problem that may occur when the decimal separator used in the file is different from the decimal separator defined by your system (a.k.a locale).
+To solve that issue, you must impose the separator using `locale = locale(decimal_mark = ",")`:
 
-## read_table()
+```{r delim-csv2ter, echo=TRUE} 
+weather3 <- read_delim(file = "Files/weather2.csv", delim = ";", locale = locale(decimal_mark = ","))
+```
+
+The last column is now recognized as `double` and the data in the object `weather3` matches the data in the file:
+``` {r comp2, echo=FALSE, fig.cap="_The data in the object now matches the data in the file._", out.width="100%"}
+knitr::include_graphics("Pics/comp2.png")
+```
+
```

Comment:
Other arguments that should be explained:
col_names, col_types, skip, n_max, skip_empty_rows
I think those are the once I use regularly.